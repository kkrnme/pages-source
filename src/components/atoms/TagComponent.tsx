import React, { ReactChild } from "react"
import { Stylable } from "../../utils/Components"
import { Link } from "gatsby"
import { PlainComponent } from "../../utils/PlainComponent"
import styled from "@emotion/styled"
import { ds, ss, cs } from "../../styles"

type TagComponentProps = Stylable & {
  children: ReactChild
}

type TagLinkProps = TagComponentProps & { children: string }

const Plain: PlainComponent<TagComponentProps> = props => (
  <label className={props.className}>{props.children}</label>
)

export const TagComponent = styled(Plain)`
  margin-left: 0.5rem;
  ${ds.padx(".5rem")}
  ${ss.rounded}
  letter-spacing: 0;
  color: ${cs.background};
  border: ${ss.border};
`

export const TagLink: React.FC<TagLinkProps> = props => (
  <TagComponent className="bg-yellow-500">
    <Link to={`/tags/${props.children}/`}>{props.children}</Link>
  </TagComponent>
)
