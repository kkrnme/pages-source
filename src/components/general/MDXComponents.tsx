import { css } from "@emotion/core"
import { Components } from "@mdx-js/react"
import React from "react"
import SwipingAnchor from "./SwipingAnchor"
import { DeepReadonly } from "ts-essentials"
import styled from "@emotion/styled"
import { cs } from "../../styles"

type Elm = JSX.IntrinsicElements

export const MDXComponents: DeepReadonly<Required<Components>> = {
  h1: styled.h1`
    border-bottom: 1px solid ${cs.border};
    font-size: 1.5rem;
    margin-top: 1rem;
    margin-bottom: 0.5rem;
  `,
  h2: styled.h2`
    border-bottom: 1px solid ${cs.border};
    font-weight: medium;
    font-size: 1.2rem;
    padding-left: 0.25rem;
    padding-right: 0.25rem;
    padding-top: 0.5rem;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
  `,
  h3: styled.h3``,
  h4: styled.h4``,
  h5: styled.h5``,
  h6: styled.h6``,
  blockquote: styled.blockquote``,
  p: styled.p`
    padding-top: 0.25rem;
    padding-bottom: 0.25rem;
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  `,
  a: (props: Elm["a"]) => {
    return props.className?.includes(`header-autolink`) ? (
      <a
        {...props}
        className="bg-monochrome-e rounded-full hover:bg-blue-400 transition mr-1"
        css={css`
          svg {
            display: inline;
          }
        `}
      />
    ) : (
      <SwipingAnchor to={props.href!} {...props} />
    )
  },
  strong: styled.strong``,
  ul: (props: Elm["ul"]) => (
    <ul className="list-disc py-1 px-2 pl-8" {...props} />
  ),
  ol: (props: Elm["ol"]) => (
    <ol className="list-decimal py-1 px-2 pl-8" {...props}></ol>
  ),
  code: (props: Elm["code"]) => <code {...props}></code>,
  delete: styled.del``,
  em: styled.em``,
  hr: styled.hr`
    margin-top: 0.75rem;
    margin-bottom: 0.75rem;
    border: 1px solid ${cs.border};
  `,
  img: styled.img``,
  inlineCode: styled.code``,
  li: styled.li``,
  pre: styled.pre``,
  thematicBreak: styled.br``,
  tr: styled.tr``,
  table: styled.table``,
  td: styled.td``,
}

export default MDXComponents
