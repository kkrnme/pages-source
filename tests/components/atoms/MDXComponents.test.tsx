import React from "react"
import renderer from "react-test-renderer"

import { MDXComponents } from "../../../src/components/atoms/MDXComponents"

for (const [k, E] of Object.entries(MDXComponents)) {
  describe(k, () => {
    it(`renders correctly: ${k}`, () => {
      const tree = renderer.create(<E>てすてす</E>).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
}
