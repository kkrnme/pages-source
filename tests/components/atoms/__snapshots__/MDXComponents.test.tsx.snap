// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a renders correctly: a 1`] = `
<span
  className="inline-block relative z-0"
>
  <a
    className="underline-anchor"
    css={
      Object {
        "map": undefined,
        "name": "cqatmp",
        "next": undefined,
        "styles": "
        color: inherit;
        transition: all 0.4s ease-in-out;
        &::before {
          display: block;
          content: \\"\\";
          height: 100%;
          background-color:#63b3ed;;
          position: absolute;
          transition: all 0.4s ease-in-out;
          top: 0;
          left: 0;
          right: 100%;
          z-index: -1;
          border-radius: 1px;
        }
        &:hover {
          color: black;
          &::before {
            left: 0;
            right: 0;
          }
        }
      ",
        "toString": [Function],
      }
    }
  >
    てすてす
  </a>
</span>
`;

exports[`blockquote renders correctly: blockquote 1`] = `
<blockquote
  className="css-0"
>
  てすてす
</blockquote>
`;

exports[`code renders correctly: code 1`] = `
<code>
  てすてす
</code>
`;

exports[`delete renders correctly: delete 1`] = `
<del
  className="css-0"
>
  てすてす
</del>
`;

exports[`em renders correctly: em 1`] = `
<em
  className="css-0"
>
  てすてす
</em>
`;

exports[`h1 renders correctly: h1 1`] = `
<h1
  className="css-1nzqhqb"
>
  てすてす
</h1>
`;

exports[`h2 renders correctly: h2 1`] = `
<h2
  className="css-sw2o99"
>
  てすてす
</h2>
`;

exports[`h3 renders correctly: h3 1`] = `
<h3
  className="css-0"
>
  てすてす
</h3>
`;

exports[`h4 renders correctly: h4 1`] = `
<h4
  className="css-0"
>
  てすてす
</h4>
`;

exports[`h5 renders correctly: h5 1`] = `
<h5
  className="css-0"
>
  てすてす
</h5>
`;

exports[`h6 renders correctly: h6 1`] = `
<h6
  className="css-0"
>
  てすてす
</h6>
`;

exports[`hr renders correctly: hr 1`] = `
<hr
  className="css-14dbf95"
>
  てすてす
</hr>
`;

exports[`img renders correctly: img 1`] = `
<img
  className="css-0"
>
  てすてす
</img>
`;

exports[`inlineCode renders correctly: inlineCode 1`] = `
<code
  className="css-0"
>
  てすてす
</code>
`;

exports[`li renders correctly: li 1`] = `
<li
  className="css-0"
>
  てすてす
</li>
`;

exports[`ol renders correctly: ol 1`] = `
<ol
  className="list-decimal py-1 px-2 pl-8"
>
  てすてす
</ol>
`;

exports[`p renders correctly: p 1`] = `
<p
  className="css-r3cwao"
>
  てすてす
</p>
`;

exports[`pre renders correctly: pre 1`] = `
<pre
  className="css-0"
>
  てすてす
</pre>
`;

exports[`strong renders correctly: strong 1`] = `
<strong
  className="css-0"
>
  てすてす
</strong>
`;

exports[`table renders correctly: table 1`] = `
<table
  className="css-0"
>
  てすてす
</table>
`;

exports[`td renders correctly: td 1`] = `
<td
  className="css-0"
>
  てすてす
</td>
`;

exports[`thematicBreak renders correctly: thematicBreak 1`] = `
<br
  className="css-0"
>
  てすてす
</br>
`;

exports[`tr renders correctly: tr 1`] = `
<tr
  className="css-0"
>
  てすてす
</tr>
`;

exports[`ul renders correctly: ul 1`] = `
<ul
  className="list-disc py-1 px-2 pl-8"
>
  てすてす
</ul>
`;
